kratos/tests/test_gid_io_gauss_points.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import print_function, absolute_import, division
+
+from KratosMultiphysics import *
+import KratosMultiphysics.KratosUnittest as UnitTest
+
+try:
+    from KratosMultiphysics.FluidDynamicsApplication import *
+    have_fluid_dynamics = True
+except ImportError:
+    have_fluid_dynamics = False
+
+import filecmp
+import os
+
+class WorkFolderScope:
+    '''Auxiliary class to define a work folder for the tests.'''
+    def __init__(self, work_folder):
+        self.currentPath = os.getcwd()
+        self.scope = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(__file__)),work_folder))
+
+    def __enter__(self):
+        os.chdir(self.scope)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        os.chdir(self.currentPath)
+
+@UnitTest.skipUnless(have_fluid_dynamics,"Missing required application: FluidDynamicsApplication")
+class TestGiDIOGaussPoints(UnitTest.TestCase):
+    '''Tests related to GiD I/O Gauss point results printing.'''
+
+    def setUp(self):
+        self.modelPart = self.setModelPart()
+        self.workFolder = "gid_io"
+
+    def tearDown(self):
+        with WorkFolderScope(self.workFolder):
+            for suffix in ['_0.post.res', '_0.post.msh']:
+                try:
+                    os.remove(self.output_file_name+suffix)
+                except FileNotFoundError:
+                    pass
+
+    def setModelPart(self):
+        modelPart = ModelPart("Test ModelPart")
+
+        modelPart.AddNodalSolutionStepVariable(DISTANCE)
+        modelPart.AddNodalSolutionStepVariable(VELOCITY)
+
+        nodes = list()
+        nodes.append( modelPart.CreateNewNode(1, 0.0, 0.0, 0.0) )
+        nodes.append( modelPart.CreateNewNode(2, 1.0, 0.0, 0.0) )
+        nodes.append( modelPart.CreateNewNode(3, 1.0, 1.0, 0.0) )
+        nodes.append( modelPart.CreateNewNode(4, 0.0, 1.0, 0.0) )
+        nodes.append( modelPart.CreateNewNode(5, 0.5, 0.5, 1.0) )
+
+        for node in nodes:
+            rx = node.X - 0.5
+            rz = node.Z - 0.5
+            r = (rx**2 + rz**2)**0.5
+            vel = Array3()
+            vel[0] = - rz/r
+            vel[1] = 0.0
+            vel[2] = rx/r
+            node.SetSolutionStepValue(VELOCITY,0,vel)
+            node.SetSolutionStepValue(DISTANCE,0,r)
+
+        properties = modelPart.GetProperties()[0]
+
+        modelPart.CreateNewElement("VMS3D",1,[1,2,4,5],properties)
```

Comment:
Hi @jcotela 
minor: why don't you use elements and conditions from the core?
This way the tests would be independent from the FluidApp

I guess you are this because otherwise you don't get results on the GPs right?

Reply:
@philbucher yes, precisely. The elements from the core do not write on gauss points.